versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/objtypes/CommitObj.java

```diff
@@ -137,7 +137,12 @@ interface Builder {
     CommitObj build();
   }
 
-  /** Creation timestamp in microseconds since epoch. */
+  /**
+   * Creation timestamp in microseconds since epoch.
+   *
+   * <p>Note that this value is not considered during Nessie commit hash value computations.
```

Comment:
Maybe worth mentioning here, because I find it sometimes confusing: this timestamp (supposedly in microseconds, but that depends on the underlying clock) is indeed not included in the hash. 

However, most commits (maybe all?) do include a header containing the so-called "commit time", coming from the commit meta: see `TypeMapping#fromCommitMeta`. That timestamp is a vanilla `Instant.now()` – so accuracy is more or less 1 ms depending on the OS.

The corollary is that for 2 commits to produce the same hash, they'd need to contain the same data incl. parents, and would need to be generated in the same "moment" – roughly, the same millisecond.

Reply:
Yep, I'll update the javadoc(s) after I resolve the related CI failures.

Reply:
Note that both timestamps can differ a lot. One is the commit-timestamp the other the timestamp when the commit was persisted (think: repo imports).